examples/barotropicqg_decaying_topography.jl

```diff
@@ -0,0 +1,233 @@
+# example numerical instability due to accumulation of enstrophy in high wavenumbers
+# is taken care with the `FilteredTimestepper` we picked. The topophic PV is prescribed
+# via keyword argument `eta`.
+prob = BarotropicQG.Problem(dev; nx=n, Lx=L, eta=topographicPV, dt=dt, stepper=stepper)
+nothing # hide
+
+# and define some shortcuts
+sol, clock, vars, params, grid = prob.sol, prob.clock, prob.vars, prob.params, prob.grid
+x, y = grid.x, grid.y
+nothing # hide
+
+# and let's plot the topographic PV:
+contourf(grid.x, grid.y, params.eta',
+          aspectratio = 1,
+            linewidth = 0,
+               levels = 10,
+                    c = :balance,
+                 clim = (-3, 3),
+                xlims = (-grid.Lx/2, grid.Lx/2),
+                ylims = (-grid.Ly/2, grid.Ly/2),
+               xticks = -3:3,
+               yticks = -3:3,
+               xlabel = "x",
+               ylabel = "y",
+                title = "topographic PV η=f₀h/H")
+
+
+# ## Setting initial conditions
+
+# Our initial condition consist of a flow that has power only at wavenumbers with
+# ``6 < \frac{L}{2\pi} \sqrt{k_x^2 + k_y^2} < 12`` and initial energy ``E_0``:
+
+E₀ = 0.04 # energy of initial condition
+
+K = @. sqrt(grid.Krsq)                          # a 2D array with the total wavenumber
+
+Random.seed!(1234)
+qih = randn(Complex{eltype(grid)}, size(sol))
+@. qih = ifelse(K < 6, 0, qih)
+@. qih = ifelse(K > 12, 0, qih)
+qih *= sqrt(E₀ / energy(qih, grid))             # normalize qi to have energy E₀
+qi = irfft(qih, grid.nx)
+
+BarotropicQG.set_zeta!(prob, qi)
+nothing #hide
+
+# Let's plot the initial vorticity field:
+
+p1 = heatmap(x, y, vars.q',
+         aspectratio = 1,
+              c = :balance,
+           clim = (-8, 8),
+          xlims = (-grid.Lx/2, grid.Lx/2),
+          ylims = (-grid.Ly/2, grid.Ly/2),
+         xticks = -3:3,
+         yticks = -3:3,
+         xlabel = "x",
+         ylabel = "y",
+          title = "initial vorticity ζ=∂v/∂x-∂u/∂y",
+     framestyle = :box)
+
+p2 = contourf(x, y, vars.psi',
+        aspectratio = 1,
+             c = :viridis,
+        levels = range(-0.35, stop=0.35, length=10), 
+          clim = (-0.35, 0.35),
+         xlims = (-grid.Lx/2, grid.Lx/2),
+         ylims = (-grid.Ly/2, grid.Ly/2),
+        xticks = -3:3,
+        yticks = -3:3,
+        xlabel = "x",
+        ylabel = "y",
+         title = "initial streamfunction ψ",
+    framestyle = :box)
+
+l = @layout Plots.grid(1, 2)
+p = plot(p1, p2, layout=l, size=(900, 400))
+
+
+# ## Diagnostics
+
+# Create Diagnostics -- `energy` and `enstrophy` functions are imported at the top.
+E = Diagnostic(energy, prob; nsteps=nsteps)
+Z = Diagnostic(enstrophy, prob; nsteps=nsteps)
+diags = [E, Z] # A list of Diagnostics types passed to "stepforward!" will  be updated every timestep.
+nothing # hide
+
+
+# ## Output
+
+# We choose folder for outputing `.jld2` files.
+filepath = "."
+filename = joinpath(filepath, "decayingbetaturb.jld2")
```

Comment:
Should you make the filename consistent with other files and remove beta reference (i.e. barotropicqg_decay_topography)